### PR TITLE
VZ-5466 Nginx and Istio update tests

### PR DESCRIPTION
**Pull request by guoyongzhang**
# Description

Acceptance tests for Nginx and Istio updates in dynamic-updates

Fixes VZ-5466

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tests/e2e/update/nginxistio/nginxistio_test.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	cr.Spec.Components.Ingress.NGINXInstallArgs = nginxInstallArgs
+	// update istio ingress
+	if cr.Spec.Components.Istio.Ingress == nil {
+		cr.Spec.Components.Istio.Ingress = &vzapi.IstioIngressSection{}
+	}
+	if cr.Spec.Components.Istio.Ingress.Kubernetes == nil {
+		cr.Spec.Components.Istio.Ingress.Kubernetes = &vzapi.IstioKubernetesSection{}
+	}
+	cr.Spec.Components.Istio.Ingress.Kubernetes.Replicas = m.istioIngressReplicas
+	// istio 1.11.4 has a bug handling this particular Affinity
```
Comment (michael-cico) [inline tests/e2e/update/nginxistio/nginxistio_test.go:56]: Do you think we need a follow-up task to track this?  I think we're going to update Istio soon, if there's a JIRA for that we could tie a test task to this case that's commented out.

File: tests/e2e/update/validation.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"fmt"
+
+	"github.com/onsi/gomega"
+	"github.com/verrazzano/verrazzano/tests/e2e/pkg"
+	v12 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ValidatePods(deployName string, labelName string, nameSpace string, expectedPodsRunning uint32, hasPending bool) {
```
Comment (michael-cico) [inline tests/e2e/update/validation.go:15]: I wonder if we could consolidate this with a utility I added for the Istio replica count checks:

https://github.com/verrazzano/verrazzano/blob/15c8321fefb4126469004d3272027d2d57ab709d/tests/e2e/pkg/kubernetes.go#L140

used from here

https://github.com/verrazzano/verrazzano/blob/master/tests/e2e/verify-install/istio/istio_test.go#L100

I think yours does some more thorough checking, might be good if they were merged.  But not a blocker for this PR.

**Review by michael-cico — APPROVED** [review]

File: tests/e2e/update/nginxistio/nginxistio_test.go (hunk 1 above)
Comment (guoyongzhang) [reply tests/e2e/update/nginxistio/nginxistio_test.go:56]: I don't think it is an important issue to worry about and to follow up.

File: tests/e2e/update/validation.go (hunk 2 above)
Comment (guoyongzhang) [reply tests/e2e/update/validation.go:15]: Agreed on merging functions in the future.

**Pull request merged**